Empty IBMCOS storage config causes operator to crashloop

When the IBMCOS storage type is used but the config itself is empty, it causes the cluster-image-registry-operator to fall into a crashloop due to NPE. The config may look like:
```
apiVersion: imageregistry.operator.openshift.io/v1
kind: Config
metadata:
  name: cluster
spec:
  logLevel: Normal
  managementState: Managed
  observedConfig: null
  operatorLogLevel: Normal
  proxy: {}
  replicas: 2
  requests:
    read:
      maxWaitInQueue: 0s
    write:
      maxWaitInQueue: 0s
  rolloutStrategy: RollingUpdate
  storage:
    ibmcos: {}
  unsupportedConfigOverrides: null
```

```
E0130 17:37:08.103637       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 483 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x26e3b40, 0x49556f0})
	/go/src/github.com/openshift/cluster-image-registry-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x7d
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x30b23b0})
	/go/src/github.com/openshift/cluster-image-registry-operator/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x75
panic({0x26e3b40, 0x49556f0})
	/usr/lib/golang/src/runtime/panic.go:1038 +0x215
github.com/openshift/cluster-image-registry-operator/pkg/storage/ibmcos.(*driver).CreateStorage(0xc00046c460, 0xc001980c00)
	/go/src/github.com/openshift/cluster-image-registry-operator/pkg/storage/ibmcos/ibmcos.go:135 +0x82
github.com/openshift/cluster-image-registry-operator/pkg/resource.(*Generator).syncStorage(0xc00063ab58, 0xc001980c00)
	/go/src/github.com/openshift/cluster-image-registry-operator/pkg/resource/generator.go:149 +0x274
github.com/openshift/cluster-image-registry-operator/pkg/resource.(*Generator).Apply(0xc001980c00, 0xc001980c00)
	/go/src/github.com/openshift/cluster-image-registry-operator/pkg/resource/generator.go:219 +0x2a
github.com/openshift/cluster-image-registry-operator/pkg/operator.(*Controller).createOrUpdateResources(0xc0009b0780, 0xc001981000)
	/go/src/github.com/openshift/cluster-image-registry-operator/pkg/operator/controller.go:205 +0xe5
github.com/openshift/cluster-image-registry-operator/pkg/operator.(*Controller).sync(0xc0009b0780)
	/go/src/github.com/openshift/cluster-image-registry-operator/pkg/operator/controller.go:259 +0x2a5
github.com/openshift/cluster-image-registry-operator/pkg/operator.(*Controller).eventProcessor.func1(0xc0009b0780, {0x25a2160, 0x30b23b0})
	/go/src/github.com/openshift/cluster-image-registry-operator/pkg/operator/controller.go:371 +0xd6
github.com/openshift/cluster-image-registry-operator/pkg/operator.(*Controller).eventProcessor(0xc0009b0780)
	/go/src/github.com/openshift/cluster-image-registry-operator/pkg/operator/controller.go:378 +0x33
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7feed8071fa8)
	/go/src/github.com/openshift/cluster-image-registry-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000ec1980, {0x30d7e00, 0xc0015c6db0}, 0x1, 0xc00010eea0)
	/go/src/github.com/openshift/cluster-image-registry-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000f717b0, 0x3b9aca00, 0x0, 0x0, 0xc000f717d0)
	/go/src/github.com/openshift/cluster-image-registry-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(0x7d7eca, 0xc0008b28b0, 0xc000f717b8)
	/go/src/github.com/openshift/cluster-image-registry-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x25
created by github.com/openshift/cluster-image-registry-operator/pkg/operator.(*Controller).Run
	/go/src/github.com/openshift/cluster-image-registry-operator/pkg/operator/controller.go:449 +0x1df
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1daf222]
```

--- a/pkg/storage/ibmcos/ibmcos.go
+++ b/pkg/storage/ibmcos/ibmcos.go
@@ -112,10 +112,10 @@ func (d *driver) UpdateEffectiveConfig() (*imageregistryv1.ImageRegistryConfigSt
 
 	var clusterLocation string
 	if infra.Status.PlatformStatus != nil {
-		if infra.Status.PlatformStatus.Type == configapiv1.IBMCloudPlatformType {
+		if infra.Status.PlatformStatus.Type == configapiv1.IBMCloudPlatformType && infra.Status.PlatformStatus.IBMCloud != nil {
 			clusterLocation = infra.Status.PlatformStatus.IBMCloud.Location
 		}
-		if infra.Status.PlatformStatus.Type == configapiv1.PowerVSPlatformType {
+		if infra.Status.PlatformStatus.Type == configapiv1.PowerVSPlatformType && infra.Status.PlatformStatus.PowerVS != nil {
 			clusterLocation, err = cosRegionForPowerVSRegion(infra.Status.PlatformStatus.PowerVS.Region)
 			if err != nil {
 				return nil, err
@@ -144,11 +144,11 @@ func (d *driver) CreateStorage(cr *imageregistryv1.Config) error {
 
 	// Set configs from Infrastructure
 	if infra.Status.PlatformStatus != nil {
-		if infra.Status.PlatformStatus.Type == configapiv1.IBMCloudPlatformType {
+		if infra.Status.PlatformStatus.Type == configapiv1.IBMCloudPlatformType && infra.Status.PlatformStatus.IBMCloud != nil {
 			d.Config.Location = infra.Status.PlatformStatus.IBMCloud.Location
 			d.Config.ResourceGroupName = infra.Status.PlatformStatus.IBMCloud.ResourceGroupName
 		}
-		if infra.Status.PlatformStatus.Type == configapiv1.PowerVSPlatformType {
+		if infra.Status.PlatformStatus.Type == configapiv1.PowerVSPlatformType && infra.Status.PlatformStatus.PowerVS != nil {
 			d.Config.Location, err = cosRegionForPowerVSRegion(infra.Status.PlatformStatus.PowerVS.Region)
 			if err != nil {
 				return err
@@ -692,10 +692,10 @@ func (d *driver) getIBMCOSClient(serviceInstanceCRN string) (*s3.S3, error) {
 
 	IBMCOSLocation := imageregistryv1.ImageRegistryConfigStorageIBMCOS{}.Location
 	if infra.Status.PlatformStatus != nil {
-		if infra.Status.PlatformStatus.Type == configapiv1.IBMCloudPlatformType {
+		if infra.Status.PlatformStatus.Type == configapiv1.IBMCloudPlatformType && infra.Status.PlatformStatus.IBMCloud != nil {
 			IBMCOSLocation = infra.Status.PlatformStatus.IBMCloud.Location
 		}
-		if infra.Status.PlatformStatus.Type == configapiv1.PowerVSPlatformType {
+		if infra.Status.PlatformStatus.Type == configapiv1.PowerVSPlatformType && infra.Status.PlatformStatus.PowerVS != nil {
 			IBMCOSLocation, err = cosRegionForPowerVSRegion(infra.Status.PlatformStatus.PowerVS.Region)
 			if err != nil {
 				return nil, err

